From: Pavlov Konstantin <thresh@altlinux.ru>
Date: Wed, 3 Jan 2007 14:05:05 +0000 (+0300)
Subject: Fix for udp format string vulnerability (VCD/CDDAX modules affected)
X-Git-Tag: 0.8.6-alt3^0~1
X-Git-Url: http://git.altlinux.ru/people/thresh/packages/?p=vlc.git;a=commitdiff_plain;h=baa3bab0cf40293ec1c2e9db7c688338d2e1ba12;hp=6697c5d3ae8cf95024c682d2c01c095f0cddbd46

Fix for udp format string vulnerability (VCD/CDDAX modules affected)
Patch by courmisch, reported by Kevin Finisterre
kfinisterre at cfm dot ohio-state dot edu .
---

Index: vlc-0.8.6/modules/access/cdda/access.c
===================================================================
--- vlc-0.8.6.orig/modules/access/cdda/access.c
+++ vlc-0.8.6/modules/access/cdda/access.c
@@ -89,17 +89,17 @@ cdio_log_handler (cdio_log_level_t level
   case CDIO_LOG_DEBUG:
   case CDIO_LOG_INFO:
     if (p_cdda->i_debug & INPUT_DBG_CDIO)
-      msg_Dbg( p_cdda_input, message);
+        msg_Dbg( p_cdda_input, "%s", message);
     break;
   case CDIO_LOG_WARN:
-    msg_Warn( p_cdda_input, message);
+    msg_Warn( p_cdda_input, "%s", message);
     break;
   case CDIO_LOG_ERROR:
   case CDIO_LOG_ASSERT:
-    msg_Err( p_cdda_input, message);
+    msg_Err( p_cdda_input, "%s", message);
     break;
   default:
-    msg_Warn( p_cdda_input, message,
+    msg_Warn( p_cdda_input, "%s\n%s %d", message,
             "the above message had unknown cdio log level",
             level);
   }
Index: vlc-0.8.6/modules/access/vcdx/access.c
===================================================================
--- vlc-0.8.6.orig/modules/access/vcdx/access.c
+++ vlc-0.8.6/modules/access/vcdx/access.c
@@ -92,17 +92,17 @@ cdio_log_handler (cdio_log_level_t level
   case CDIO_LOG_DEBUG:
   case CDIO_LOG_INFO:
     if (p_vcdplayer->i_debug & INPUT_DBG_CDIO)
-      msg_Dbg( p_vcd_access, message);
+      msg_Dbg( p_vcd_access, "%s", message);
     break;
   case CDIO_LOG_WARN:
-    msg_Warn( p_vcd_access, message);
+    msg_Warn( p_vcd_access, "%s", message);
     break;
   case CDIO_LOG_ERROR:
   case CDIO_LOG_ASSERT:
-    msg_Err( p_vcd_access, message);
+    msg_Err( p_vcd_access, "%s", message);
     break;
   default:
-    msg_Warn( p_vcd_access, message,
+    msg_Warn( p_vcd_access, "%s\n%s %d", message,
             _("The above message had unknown log level"),
             level);
   }
@@ -118,14 +118,14 @@ vcd_log_handler (vcd_log_level_t level, 
   case VCD_LOG_DEBUG:
   case VCD_LOG_INFO:
     if (p_vcdplayer->i_debug & INPUT_DBG_VCDINFO)
-      msg_Dbg( p_vcd_access, message);
+      msg_Dbg( p_vcd_access, "%s", message);
     break;
   case VCD_LOG_WARN:
-    msg_Warn( p_vcd_access, message);
+    msg_Warn( p_vcd_access, "%s", message);
     break;
   case VCD_LOG_ERROR:
   case VCD_LOG_ASSERT:
-    msg_Err( p_vcd_access, message);
+    msg_Err( p_vcd_access, "%s", message);
     break;
   default:
     msg_Warn( p_vcd_access, "%s\n%s %d", message,
